sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -190,9 +190,12 @@ MemObjRecord *Scheduler::GraphBuilder::getOrInsertMemObjRecord(
         DepDesc Dep = findDepForRecord(Dependant, Record);
         Dep.MDepCommand = Dependency;
         std::vector<Command *> ToCleanUp;
-        if (Command *ConnectionCmd = Dependant->addDep(Dep, ToCleanUp))
+        Command *ConnectionCmd = Dependant->addDep(Dep, ToCleanUp);
+        if (ConnectionCmd)
           ToEnqueue.push_back(ConnectionCmd);
-        Dependency->addUser(Dependant);
+        else
+          Dependency->addUser(Dependant);
```

Comment:
A comment will be helpful justifying why `Dependant` is added as user of `Dependency` only when there's no connection command.

This is a common change to several places. `addUser` comes along with `addDep`. Now, it has some sophisticated logic. Can we have it wrapped into `addDep` or a distinct helper, possibly inline?

Reply:
it seems to be not possible to wrap it to addDep because in some spaces I haven't found addUser pair for addDep call (for release commands for example). For extra helper - ok, will do it.